### ISSUE TITLE
`__rust_probestack` test in `rust_example.rs` is causing a stack overflow

### ISSUE THREAD

**Issue by wedsonaf** [Rust-for-Linux/linux#111]
The stack size is 8KB, having a local variable of 8KB+32 bytes takes it over the edge.

I have seen failures on debug builds for this, for example,  https://github.com/Rust-for-Linux/linux/runs/2107043283?check_suite_focus=true -- it has the following error:

```
[   13.127113] Large array has length: 1028
[   13.131438] BUG: stack guard page was hit at (____ptrval____) (stack is (____ptrval____)..(____ptrval____))
[   13.131438] kernel stack overflow (double-fault): 0000 [#1] SMP DEBUG_PAGEALLOC KASAN PTI
```

For now I'm reducing the variable size in #104 to work around this. Please fix if you need it to be bigger to ensure probestack is properly tested.

**Comment by alex** [comment]
Maybe we should put it in a different test file so that it doesn't disrupt code that's expected to work :-)

**Comment by ojeda** [comment]
Indeed. Reorganizing all the test code is something I will work on, but first I need to finish setting up the "independent CI".

Nevertheless, it is good we are seeing the error in the debug builds, it means debug CI is being useful ;-)

**Comment by adamrk** [comment]
So this was triggered by an upstream merge from https://github.com/Rust-for-Linux/linux/pull/112. Does anyone know why (just trying to understand myself)?

**Comment by ojeda** [comment]
The CI link mentioned by Wedson happened before #112 -- it is not a particular change: the frame size has been growing with the added test cases, which means we are crossing the tipping point now.

Eventually we will move these into proper tests, but for the moment my plan is moving them into individual kernel modules since anyhow I need to clean up all the GitHub-only bits (which include `rust_example.rs`) to be able to get into linux-next properly.

**Comment by adamrk** [comment]
Got it, thanks!
